### PR TITLE
Add release guide doc

### PR DESCRIPTION
**Pull request by kezhenxu94**
...and include the LICENSE, NOTICE, documentation files in binary tar

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -15,7 +15,7 @@
 # limitations under the License.
 #
 
-VERSION ?= latest
+VERSION ?= 0.1.0
```
Comment (wu-sheng) [inline Makefile]: I think we should set it to `0.2.0-latest`? And with a real release script to override this?

In traditional, we should follow the pattern, version + snapshot/latest/commit id as dev version. 0.1.0， 0.2.0 should only work for release, right?

Comment (kezhenxu94) [reply Makefile]: > I think we should set it to 0.2.0-latest?

> In traditional, we should follow the pattern, version + snapshot/latest/commit id as dev version. 0.1.0， 0.2.0 should only work for release, right?

I prefer `0.2.0-snapshot`

> And with a real release script to override this?

Now the `makefile` serves as release script, if we add a release script, then the version in that script should be updated in every release? say `sh release.sh --version=0.2.0`, I don't think it's different from `make VERSION=0.2.0 release`.

Comment (wu-sheng) [reply Makefile]: My point is, when it is not release tag, it should package the codes as 0.2.0-snapshot. When the release happens, people should input
1. GitHub user name
1. Github password
1. Tag name
1. Release version
1. Next version

Then all things should be changed, and tag after those, push tag and git commits to the branch. Then move the version to the next version. So there are 3 git push
1. Version change to stable version(no snapshot) as commit 1
1. Tag this commit and push
1. Version changes to the next snapshot as commit 2

You should have used this in the main repo release, right? This is my expected release script. Make sense?

Comment (kezhenxu94) [reply Makefile]: Uh I got your point now, you're expecting the release process to be more automated, make sense to me absolutely

Comment (wu-sheng) [reply Makefile]: > Uh I got your point now, you're expecting the release process to be more automated, make sense to me absolutely

Yes. I hope more people are able to be the release manager. As this project is not like the main repo, people are facing the network issue.

**Comment by wu-sheng** [comment]
Should we close this?

**Comment by kezhenxu94** [comment]
> Should we close this?

@wu-sheng Let's keep it opened for a while, I have optimized the `Makefile` according to the consensus that we had reached:

> Uh I got your point now, you're expecting the release process to be more automated, make sense to me absolutely

**Comment by kezhenxu94** [comment]
@wu-sheng Actually I think we made it too complicated, a release guild documentation is totally enough for this project, it is not a maven project and it doesn't rely on the version number strictly, (maven modules are version-dependent).

The release process is rather simple and straightforward now, you can see below, there are only 3 steps and no new commits are required (only pushing tag is needed), and I don't see any benefit to bring the version number in the `Makefile`

- Export the version that is to be released, `export VERSION=1.0.1 `
- Tag the latest commit that is to be released with `git tag v${VERSION}` and push the tag with `git push https://github.com/apache/skywalking-cli v${VERSION}`
- Verify licenses, build and sign distribution packages, simply run `make release`, distribution packages and checksums are generated
- [Upload the packages to SVN repository](https://github.com/apache/skywalking/blob/master/docs/en/guides/How-to-release.md#upload-to-apache-svn) 
- [Send internal announcement](https://github.com/apache/skywalking/blob/master/docs/en/guides/How-to-release.md#make-the-internal-announcements)
- [Wait at least 48 hours for test responses](https://github.com/apache/skywalking/blob/master/docs/en/guides/How-to-release.md#wait-at-least-48-hours-for-test-responses)
- [Call for vote](https://github.com/apache/skywalking/blob/master/docs/en/guides/How-to-release.md#call-a-vote-in-dev)
- [Publish release](https://github.com/apache/skywalking/blob/master/docs/en/guides/How-to-release.md#publish-release)

**Comment by wu-sheng** [comment]
I am OK with this release steps, my key point is that, when, who and how we should change the `Makefile` file, which includes a version number. https://github.com/apache/skywalking-cli/pull/23/files#r355111578

**Comment by kezhenxu94** [comment]
> I am OK with this release steps, my key point is that, when, who and how we should change the `Makefile` file, which includes a version number. https://github.com/apache/skywalking-cli/pull/23/files#r355111578

We may never need to change the version number in the `Makefile`

**Comment by wu-sheng** [comment]
Then why we have it in the file? And we have cli command about version, how to change that?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -81,4 +81,22 @@ and run the tests locally, make sure all tests passed.
...
+
+### Release steps
+- Export the version that is to be released, `export VERSION=1.0.1 `
+- Tag the latest commit that is to be released with `git tag v${VERSION}` and push the tag with `git push https://github.com/apache/skywalking-cli v${VERSION}`
+- Verify licenses, build and sign distribution packages, simply run `make release`, distribution packages and checksums are generated
+- [Upload the packages to SVN repository](https://github.com/apache/skywalking/blob/master/docs/en/guides/How-to-release.md#upload-to-apache-svn) 
+- [Send internal announcement](https://github.com/apache/skywalking/blob/master/docs/en/guides/How-to-release.md#make-the-internal-announcements)
+- [Wait at least 48 hours for test responses](https://github.com/apache/skywalking/blob/master/docs/en/guides/How-to-release.md#wait-at-least-48-hours-for-test-responses)
+- [Call for vote](https://github.com/apache/skywalking/blob/master/docs/en/guides/How-to-release.md#call-a-vote-in-dev)
+- [Publish release](https://github.com/apache/skywalking/blob/master/docs/en/guides/How-to-release.md#publish-release)
```
Comment (wu-sheng) [inline CONTRIBUTING.md]: Also, you ref the main doc, it isn't totally right, Such as the svn path, mail contents. I suggest you copy them and change them right.

**Comment by kezhenxu94** [comment]
> Then why we have it in the file? And we have cli command about version, how to change that?

I behaves as a placeholder, making it possible to pass a variable in command line to replace the version

**Comment by kezhenxu94** [comment]
@wu-sheng I think the release process is easy and clear enough, please take a look at the docs

File: docs/How-to-release.md [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+All PMC members and committers should check these before voting +1.
+
+1. Features test.
+1. All artifacts in staging repository are published with `.asc`, `.md5`, and `sha` files
+1. Source codes and distribution packages (`apache-skywalking-cli-{src,bin}-$VERSION.tgz`)
+are in `https://dist.apache.org/repos/dist/dev/skywalking/cli/$VERSION` with `.asc`, `.sha512`.
+1. `LICENSE` and `NOTICE` are in source codes and distribution package.
+1. Check `shasum -c apache-skywalking-cli-{src,bin}-$VERSION.tgz.sha512`
+1. Build distribution from source code package by following this [the build guide](#build-and-sign-the-source-code-package).
+1. Apache RAT check.
```
Comment (wu-sheng) [inline docs/How-to-release.md]: I think you don't have rat check, yes?

File: Makefile (hunk 1 above)
Comment (wu-sheng) [inline Makefile]: I think you said `I prefer 0.2.0-snapshot`?

Comment (wu-sheng) [reply Makefile]: Or keep latest?

File: docs/How-to-release.md (hunk 3 above)
Comment (kezhenxu94) [reply docs/How-to-release.md]: > I think you don't have rat check, yes?

Actually we have, docs are updated with the command

File: Makefile (hunk 1 above)
Comment (kezhenxu94) [reply Makefile]: agree

**Comment by wu-sheng** [comment]
I think you have a license check, not rat. Could you point to me where the rat is?

**Comment by kezhenxu94** [comment]
> I think you have a license check, not rat. Could you point to me where the rat is?

Well, I've been mistaken "rat" for "license check", that's what I mean "actually we have", we have license checks, but if you insist that it's not "rat", I can change the words in the docs, to "license check"

image

> picture from https://creadur.apache.org/rat/

**Comment by kezhenxu94** [comment]
Updated

**Pull request merged**